src/components/statistic/statistic.tsx

```diff
@@ -0,0 +1,122 @@
+import { CardComponent } from "@components/shared/card/card";
+import { KeyValueComponent } from "@components/shared/key-value-list/key-value";
+import { KeyValueListComponent } from "@components/shared/key-value-list/key-value-list";
+import { ModalComponent } from "@components/shared/modal/modal";
+import { ValueSelectorComponent } from "@components/shared/value-selector/value-selector";
+import { computed, signal } from "@preact/signals";
+import { getWebAppData } from "@utils/telegram.utils";
+import { Fragment, useContext } from "preact/compat";
+import {
+  GameStateContext,
+  StatisticContext,
+  ThemeContext,
+} from "../../interfaces/context";
+import "./statistic.css";
+
+const debugClickCount = signal(0);
+const isDebugActive = computed(() => {
+  return debugClickCount.value > 3;
+});
+
+export function StatisticComponent() {
+  const {
+    averageCardFlipsCount,
+    averageTimeSpentInSeconds,
+    gameLevelForStatistic,
+    increaseStatisticLevel,
+    degreesStatisticLevel,
+    currentLevelStatistic,
+  } = useContext(StatisticContext);
+  const { toggleTheme, theme } = useContext(ThemeContext);
+  const {
+    increaseLevel,
+    degreesLevel,
+    gameLevel,
+    showDebugInfo,
+    toggleShowDebugInfo,
+  } = useContext(GameStateContext);
```

Comment:
To keep the context simpler, you might want to retain only the state there and import action creators directly or via a custom hook.